device/src/key_generator.rs

```diff
@@ -0,0 +1,23 @@
+// TODO: Use the HMAC peripheral
```

Comment:
This is just a placeholder for using the HMAC peripheral. I took a small attempt at doing this but I need @musdom's advice first.